crates/payload/basic/src/lib.rs

```diff
@@ -425,6 +447,39 @@ where
                 let _ = tx.send(res);
             }));
 
+            // In Optimism, the PayloadAttributes can specify a `no_tx_pool` option that implies we
+            // should not pull transactions from the tx pool. In this case, we build the payload
```

Comment:
When is `no_tx_pool` set from OP Node?

Reply:
When pulling batches from L1 and recreating the execution payloads from them, this makes it deterministic